clients/drcachesim/common/trace_entry.h

```diff
@@ -442,13 +453,38 @@ typedef enum {
      */
     TRACE_MARKER_TYPE_SYSCALL,
 
+    /**
+     * This marker is emitted prior to a system call which is known to block under some
+     * circumstances.  Whether it blocks may depend on the values of parameters or the
+     * state of options set in prior system calls.  If these markers are present, the
+     * file type #OFFLINE_FILE_TYPE_KERNEL_SCHED is set.  The marker value is 0.
+     */
+    TRACE_MARKER_TYPE_MAYBE_BLOCKING_SYSCALL,
+
     // ...
     // These values are reserved for future built-in marker types.
     // ...
     TRACE_MARKER_TYPE_RESERVED_END = 100,
     // Values below here are available for users to use for custom markers.
 } trace_marker_type_t;
 
+/** Constants related to function or system call parameter tracing. */
+enum {
+/**
+ * When system call parameter and return values are provided, they use the function
+ * tracing markers #TRACE_MARKER_TYPE_FUNC_ID, #TRACE_MARKER_TYPE_FUNC_ARG, and
+ * #TRACE_MARKER_TYPE_FUNC_RETVAL.  The identifier used for
+ * #TRACE_MARKER_TYPE_FUNC_ID is equal to this base value plus the 32-bit system
+ * call number for 64-bit marker values or this base value plus the lower 16 bits
+ * of the system call number for 32-bit marker values.
```

Comment:
"lower 16 bits of the system call number": are syscall numbers larger than that at all?

Reply:
Yes, in multiple OS variants upper bits are set to indicate various things.  DR removes some Mac upper bits but there are others -- e.g. see some of the WOW64 numbers: https://github.com/DynamoRIO/drmemory/blob/master/drsyscall/drsyscall_numx.h#L2115